stablehlo/reference/Ops.cpp

```diff
@@ -224,6 +225,47 @@ Tensor evalTransposeOp(const Tensor &operand, ArrayRef<int64_t> permutation,
   return result;
 }
 
+// ***********************************************************************
+// This is an simplified implementation of while op semantics. The
+// simplification is based on iterating the loop `dummy_limit` number of times
+// if the loop condition is `true`. The simplifiction is done as compare op is
+// not supported. To be corrected and improved as part of #967 and #992
+// respectively.
+// ***********************************************************************
+SmallVector<Tensor> evalWhileOp(ArrayRef<Tensor> runtimeInputs, Region &cond,
+                                Region &body, const InterpreterScope &scope) {
+  SmallVector<Tensor> runtimeResults(runtimeInputs);
+  bool isIsolatedFromAbove =
+      cond.getParentOp()->hasTrait<OpTrait::IsIsolatedFromAbove>();
+
+  auto runtimeCondResultsOrErr =
+      eval(cond, runtimeInputs, isIsolatedFromAbove ? nullptr : &scope);
+  if (!runtimeCondResultsOrErr)
+    llvm::report_fatal_error("Error in while op evaluation");
```

Comment:
Error reporting is pretty underdeveloped in the interpreter at the moment, but let's maybe make these messages more specific? E.g. here we could say "Failed to evaluate cond", immediately below we could say "Failed to evaluate body".